Add dropzone to remove events from team planner

oliverguenther opened the pull request:
- [x] When a user starts dragging a work package, along with the hovered date ranges on the calendar being highlighted in blue, a drag zone appears at the bottom, to the right of the "+ Add assignee" button.
- [x] The drag zone spans the full available width to the right.
- [x] It contains this hint: "Drag here to remove assignee and start and end dates."
- [x] When a work package is dragged to this zone, the state of the zone changes (blue fill) to communicate that the drag zone has an acceptable work package, and that dropping it there will perform an action.
- [x] On mouse out, the work package is "removed" from the current view (it disappears) and a green success toast informs the user: ~~"Assignee and start/finish dates successfully removed."~~ Non-specific "Update successful." message kept as discussed
- ~~The work package becomes available again via the left-hand add existing search pane.~~ Extracted into [OP#40988](https://community.openproject.org/wp/40988)
- [x] Fix dragging of new cards with angular templates, as they are currently being destroyed when dragged outside team planner.

[OP#40883](https://community.openproject.org/wp/40883)

oliverguenther committed:
Add dropzone to remove events from team planner

b12f commented on frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.html:
```diff
@@ -56,18 +56,46 @@
+      ></span>
+        <span
+          *ngIf="!dropzone.canDrop"
+          class="op-team-planner--dropzone-label op-team-planner--dropzone-label_forbidden"
```
I don't think you need the extra forbidden class here if it's on the parent already

oliverguenther committed:
Detach views instead of removing them immediately

Fullcalendar actively removes the DOM elements from our rendered
template in https://github.com/fullcalendar/fullcalendar/blob/master/packages/common/src/global-plugins.ts#L80-L85

This results in the view still being usable, but our component has gone.

oliverguenther committed:
Fix external dropping by setting an ID
